### DISCUSSION TITLE
Couldn't make remark-slug work

### DISCUSSION THREAD

**Discussion by hakankaraduman** [remarkjs/remark#627] [Q&A] [answered]
Hi,

I added remark-slug to my code as below

```
<React.Fragment>
      <Head>
        <title>{doc.title}</title>
      </Head>
      <div className="markdown">
        <ReactMarkdown
          plugins={[[gfm, { singleTilde: false }], directive, slug]}
          source={doc.markdown}
          renderers={{
            code: CodeHighlighter,
            containerDirective: ContainerDirective,
          }}
        />
      </div>
    </React.Fragment>
```

But the ```<h1>``` etc. elements don't have id still. I use next.js. Does anybody know the reason or how do I debug this? Is there a way to inject code in to the process?

To give more information, I also create table of contents with "mdast-util-from-markdown" package and I want these toc items to keep track of current scroll and highliht. That's why I need the ids and slug of headings.

**Comment by ChristianMurphy** [comment, marked as answer]
It's a known issue https://github.com/remarkjs/react-markdown/issues/397
There is a fix in the work https://github.com/remarkjs/react-markdown/issues/397#issuecomment-705814332

An alternative in the meantime, is another package built by the remark/unified team https://github.com/remarkjs/react-remark which does support remark-slug

**Reply by hakankaraduman** [reply]
Thanks for the reply, do you know how to feed these renderers to react-remark? I also need to use remark-directive and remark-gfm

```
renderers={{
  code: CodeHighlighter,
  containerDirective: ContainerDirective,
}}
```

**Reply by hakankaraduman** [reply]
I fixed my issue by updating html elements with slug from github-slugger on useEffect hook. 
Still would love to learn how to use those other plugins with react-remark. Thanks

**Reply by ChristianMurphy** [reply]
Renderers in `react-remark` are part of `rehypeReactOptions.components`, example here: https://github.com/remarkjs/react-remark#pass-options-to-component
`code` would still be `code`, Directives would need a `passthrough` on `remarkToRehypeOptions`.
